internal/indexmanagement/reconcile.go

```diff
@@ -292,17 +282,25 @@ func newCronJob(clusterName, image, namespace, name, schedule string, nodeSelect
 		Env: []corev1.EnvVar{
 			{Name: "ES_SERVICE", Value: fmt.Sprintf("https://%s:9200", clusterName)},
 		},
+		Command: []string{"bash"},
+		Args: []string{
+			"-c",
+			scriptPath,
+		},
 	}
 	container.Env = append(container.Env, envvars...)
-	fnContainerHander(&container)
 
 	container.VolumeMounts = []corev1.VolumeMount{
 		{Name: "certs", ReadOnly: true, MountPath: "/etc/indexmanagement/keys"},
 		{Name: "scripts", ReadOnly: false, MountPath: "/tmp/scripts"},
 	}
+	return &container
+}
+
+func newCronJob(clusterName, namespace, name, schedule string, nodeSelector map[string]string, tolerations []corev1.Toleration, initContainers []*corev1.Container) *batch.CronJob {
 	podSpec := corev1.PodSpec{
 		ServiceAccountName: clusterName,
-		Containers:         []corev1.Container{container},
+		InitContainers:     []corev1.Container{},
```

Comment:
why are we using init containers?

Reply:
Ref comment 0: `One risk in this proposed solution is failure of the first container will block any subsequent behaviors. Placing the PR on hold pending agreement its a viable solution`

Init containers run sequentially which is what we want; we want to stay out of a race condition where we are rolling over and moving the write alias but also trying to remove indices.  Regular containers would run in parallel.  The downside to this approach is the running of the second container (rollover) is always contingent upon the successful completion of delete.  I'm not convinced this is necessarily the best approach but was the quickest path to a change without modifying any scripts.  It could be we are better served accepting a "delete" failure and then just continuing to try and rollover so that both continue to happen.

Reply:
i think we want to move away from the init container approach. can't we update our scripts to be blocking and not require the prior action to pass? (e.g. if delete fails we will still try to roll over)

Reply:
Init containers are a convenient mechanism to isolate specific tasks and I believe it to be a good application here aside from the need to possibly run task 2 regardless of task 1.  I will pose this question before making mods because I don't know the best answer:  is there a drawback to continuing to rollover if delete were to fail?

Issue i could see is we add more shards thus consuming more RAM in lieu of simply allowing the index to increase in size.  That likely has its own limitations but it may be lesser issue.  I defer to @lukas-vlcek here to provide some feedback

Reply:
if we don't rollover we risk the index being written to becoming large and exceeding recommended sizes

Reply:
per our discussion, move logic from init containers to single container running scripts